bwt/wavelet.go

```diff
@@ -0,0 +1,356 @@
+package bwt
+
+import (
+	"fmt"
+	"math"
+
+	"golang.org/x/exp/slices"
+)
+
+/*
+
+For the waveletTree's usage, please read the its
+method documentation. To understand what it is and how
+it works for either curiosity or maintenance, then read below.
+
+# WaveletTree
+
+The Wavelet Tree allows us to conduct RSA queries on strings. in
+a memory and run time efficient manner.
+RSA stands for (R)ank, (S)elect, (A)ccess.
```

Comment:
```suggestion
RSA stands for (R)ank, (S)elect, (A)ccess.

See this blog post by Alex Bowe for an additional explanation:
https://www.alexbowe.com/wavelet-trees/
```
I found this explanation's visuals very helpful, I'd recommend including it.